packages/core/src/lighting/DirectLight.ts

```diff
@@ -21,6 +23,7 @@ export class DirectLight extends Light {
   static _updateShaderData(shaderData: ShaderData): void {
     const data = DirectLight._combinedData;
 
+    shaderData.setIntArray(DirectLight._cullingMaskProperty, data.cullingMask);
```

Comment:
why not use `setFloat` instead using 2 mask in IntArray?

Reply:
https://www.khronos.org/registry/OpenGL/specs/es/2.0/GLSL_ES_Specification_1.00.pdf

Integer precision qualifiers specify the desired precision of integer data types used in the declaration of variables. They specify a minimum range and precision supported by the implementation for each precision qualifier. The available integer precision qualifiers are:

highp: represents high precision, supports at least 16 bits of integer precision.
mediump: represents medium precision, supports at least 10 bits of integer precision.
lowp: represents low precision, supports at least 8 bits of integer precision.

Reply:
To be honest, I don't want to be compatible with webgl1.0!